src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx

```diff
@@ -7,22 +7,31 @@ import { encodeMultiSendData } from '@gnosis.pm/safe-core-sdk/dist/src/utils/tra
 import { useWeb3 } from '@/hooks/wallets/web3'
 import { Button, DialogContent, Typography } from '@mui/material'
 import EthHashInfo from '@/components/common/EthHashInfo'
-import { useMemo, useState } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 import { dispatchBatchExecution } from '@/services/tx/txSender'
 import { generateDataRowValue } from '@/components/transactions/TxDetails/Summary/TxDataRow'
 import { Errors, logError } from '@/services/exceptions'
 import ErrorMessage from '@/components/tx/ErrorMessage'
 import { BatchExecuteData } from '@/components/tx/modals/BatchExecuteModal/index'
 import DecodedTxs from '@/components/tx/modals/BatchExecuteModal/DecodedTxs'
 import { getMultiSendTxs, getTxsWithDetails } from '@/utils/transactions'
+import { trackEvent, TX_LIST_EVENTS } from '@/services/analytics'
 
 const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubmit: (data: null) => void }) => {
   const [isSubmittable, setIsSubmittable] = useState<boolean>(true)
   const [submitError, setSubmitError] = useState<Error | undefined>()
   const chain = useCurrentChain()
-  const { safe, safeAddress } = useSafeInfo()
+  const { safe } = useSafeInfo()
   const provider = useWeb3()
 
+  // Track the number of transactions in the batch
+  useEffect(() => {
+    trackEvent({
+      ...TX_LIST_EVENTS.BATCH_EXECUTE,
```

Comment:
This is already being tracked in `BatchExecuteButton`.

Reply:
Thanks, removed!